db/compaction/compaction_job.cc

```diff
@@ -971,16 +980,19 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
     if (c_iter->status().IsManualCompactionPaused()) {
       break;
     }
-    if (!output_file_ended && c_iter->Valid() &&
-        sub_compact->compaction->output_level() != 0 &&
-        sub_compact->ShouldStopBefore(c_iter->key(),
-                                      sub_compact->current_output_file_size) &&
-        sub_compact->builder != nullptr) {
-      // (2) this key belongs to the next file. For historical reasons, the
-      // iterator status after advancing will be given to
-      // FinishCompactionOutputFile().
-      input_status = input->status();
-      output_file_ended = true;
+    if (!output_file_ended && c_iter->Valid()) {
+      if (((partitioner.get() != nullptr &&
+            partitioner->ShouldPartition(c_iter->key())) ||
```

Comment:
Is the key passing to `partitioner` an user key or internal key? I think it is better to let the `partitioner` only deal with user key.

Reply:
It is user key. Internal key would be c_iter->ikey();

Reply:
There's also `c_iter->user_key()`. I think both `c_iter->key()` and `c_iter->ikey()` are internal key, except the `ikey` is in parsed form.

Reply:
done